### PR TITLE
Don't show spinner with verbose output

### LINKED ISSUE #146: Disable spinner in debugging mode
Spinner is a bit annoying when debugging is enabled.
```sh
$ aiidalab-launch -vv start
[Verbose logging is enabled. (LEVEL=20)
INFO:aiidalab_launch:Configuration file path: /home/hollas/.config/org.aiidalab.aiidalab_launch/config.toml
INFO:aiidalab_launch:Using profile: test
INFO:aiidalab_launch:Starting container 'aiidalab_test'...
Starting container... |INFO:aiidalab_launch:Started container: aiidalab_test (42e544244d4d254b1db30e6806f2756d2ae3167ca3d74e118a1bc5cb1816726a).
done.
Waiting for AiiDAlab instance to get ready... -INFO:aiidalab_launch:Waiting for services to come up (42e544244d4d254b1db30e6806f2756d2ae3167ca3d74e118a1bc5cb1816726a)..
```

I very briefly looked at the code but it wasn't immediately clear how to disable it.

### PR DESCRIPTION
**Pull request by danielhollas**
When verbose output is enabled via the `-v` option, the spinner interferes with the extra output.

Closes #146

### CODE REVIEW

File: aiidalab_launch/util.py [hunk 1]
```diff
@@ -43,6 +44,13 @@ def spinner(
     def spin() -> None:
         if msg:
             click.echo(f"{msg.rstrip()} ", nl=False, err=True)
+
+        # Don't show spinner if verbose output is enabled
+        level = logging.getLogger().getEffectiveLevel()
+        if level != 0 and level < 40:
```
Comment (yakutovicha) [inline aiidalab_launch/util.py:50]: I am not very experienced with this particular code, but the `level < 40` condition seems strange. If a higher number means "more verbose", then there should be no upper limit, no?

**Review by yakutovicha — COMMENTED** [review]
One little question.

File: aiidalab_launch/util.py (hunk 1 above)
Comment (unkcpz) [reply aiidalab_launch/util.py:50]: @danielhollas Is the user allowed to change the level? If I understand correctly, the level is increased by adding more `v`? If this is the case, it is better to update the help information of `-v`.

Comment (danielhollas) [reply aiidalab_launch/util.py]: @yakutovicha actually, the level indicates severity, so lower number means less severe, hence more verbose. See

https://docs.python.org/3/library/logging.html#logging-levels

The default level is `logging.ERROR` which is 40. I have replaced the hardcoded numbers with the proper variables from the logging module, thanks for pointing this out.

Comment (danielhollas) [reply aiidalab_launch/util.py]: > Is the user allowed to change the level? If I understand correctly, the level is increased by adding more v? If this is the case, it is better to update the help information of -v.

@unkcpz yes, you can adjust the level based on the number of `v`s. The current help says

> Provide this option to increase the output verbosity of the launcher.

@yakutovicha @unkcpz do you have suggestion how to improve this? I cannot figure out how to word this concisely.

Comment (unkcpz) [reply aiidalab_launch/util.py]: I am sure no good at wording docs. I check the python cmdline doc https://docs.python.org/3/using/cmdline.html#cmdoption-1. We can assume the user only uses `-v` and `-vv` and for `-vv` giving the detail of what will be printed more.

File: aiidalab_launch/util.py [hunk 2]
```diff
@@ -41,13 +42,24 @@ def spinner(
...
-            click.echo(f"{msg.rstrip()} ", nl=False, err=True)
-        with click_spinner.spinner():  # type: ignore
+            newline = not show_spinner
+            click.echo(f"{msg.rstrip()} ", nl=newline, err=True)
+
+        if show_spinner:
+            with click_spinner.spinner():  # type: ignore
+                stop.wait()
+            click.echo(
+                (final or "done.") if (completed.is_set() and msg) else " ", err=True
```
Comment (yakutovicha) [inline aiidalab_launch/util.py:59]: I believe the line could be slightly simplified, if the default value for `final` was set to `"done."`
```suggestion
                final if (completed.is_set() and msg) else " ", err=True
```

**Review by yakutovicha — COMMENTED** [review]
Almost good to go for me. Just one little suggestion.

File: aiidalab_launch/util.py (hunk 2 above)
Comment (danielhollas) [reply aiidalab_launch/util.py]: Good idea. Done. :-)

**Review by yakutovicha — APPROVED** [review]

**Pull request merged**